GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+                    else if (module.InTheMiddleOfPatch())
+                    {
+                        await SetStateAsync(State.Patch, conflicted);
+                    }
+                    else
+                    {
+                        await SetStateAsync(State.None, false);
+                    }
+                }
+            }
+
+            async Task SetStateAsync(State state, bool conflicted)
+            {
+                if ((state == _state) && (conflicted == _hasConflicts))
+                {
+                    // nothing to do
+                    return;
+                }
+
+                _state = state;
+                _hasConflicts = conflicted;
+
+                // now update in main UI thread
+                await this.SwitchToMainThreadAsync();
+
+                // remove old controls
+                ButtonContainer.Controls.Clear();
+
+                string action = "";
+
+                switch (_state)
+                {
+                    case State.Bisect:
+                        action = "Bisect";
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Rebase:
+                        action = "Rebase";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                    case State.Merge:
+                        action = "Merge";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        break;
+                    case State.Patch:
+                        action = "Patch";
+                        ButtonContainer.Controls.Add(_hasConflicts ? ResolveButton : ContinueButton);
+                        ButtonContainer.Controls.Add(AbortButton);
+                        ButtonContainer.Controls.Add(MoreButton);
+                        break;
+                }
+
+                if (_state != State.None)
+                {
+                    string text = action + " currently in progress";
+                    if (_hasConflicts)
+                    {
+                        text += " with merge conflicts";
+                    }
+
+                    BackColor = _hasConflicts ? System.Drawing.Color.DarkOrange : System.Drawing.Color.LightSkyBlue;
+
+                    Label.Text = text;
+                    Show();
+                }
+                else
+                {
+                    Hide();
+                }
+            }
+        }
+
+        private void ResolveButton_Click(object sender, EventArgs e)
+        {
+            GitModuleForm form = FindForm() as GitModuleForm;
+            form.UICommands.StartResolveConflictsDialog(this);
```

Comment:
❗️ Line:131 carries a risk of `form` being `null`.
```suggestion
            form?.UICommands.StartResolveConflictsDialog(this);
```